lib/spack/spack/install_test.py

```diff
@@ -468,12 +467,11 @@ def write_tested_status(self):
 def test_part(pkg: Pb, test_name: str, purpose: str, work_dir: str = ".", verbose: bool = False):
     wdir = "." if work_dir is None else work_dir
     tester = pkg.tester
-    # TODO (post-34236): "test"->"test_" once remove deprecated methods
     assert test_name and test_name.startswith(
-        "test"
-    ), f"Test name must start with 'test' but {test_name} was provided"
+        "test_"
+    ), f"Test name must start with 'test_' but {test_name} was provided"
 
```

Comment:
This should be removed.  No-one will have `def test_(self, ...)` defined to warn them about.